components/eamxx/src/share/io/scream_output_manager.cpp

```diff
@@ -299,6 +300,24 @@ void OutputManager::init_timestep (const util::TimeStamp& start_of_step, const R
     return;
   }
 
+  // Make sure dt is in the control
+  m_output_control.set_dt(dt);
+
+  if (start_of_step==m_case_t0 and m_avg_type==OutputAvgType::Instant and
+      m_output_file_specs.storage.type!=NumSnaps and m_output_control.frequency_units=="nsteps") {
+    // This is the 1st step of the whole run, and a very sneaky corner case. Bear with me.
+    // When we call run, we also compute next_write_ts. Then, we use next_write_ts to see if the
+    // next output step will fit in the currently open file, and, if not, close it right away.
+    // For a storage type!=NumSnaps, we need to have a valid timestamp for next_write_ts, which
+    // for freq=nsteps requires to know dt. But at t=case_t0, we did NOT have dt, which means we
+    // computed next_write_ts=last_write_ts (in terms of date:time, the num_steps is correct).
+    // This means that at that time we deemed that the next_write_ts definitely fit in the same
+    // file as last_write_ts (date/time are the same!), which may or may not be true for non NumSnaps
+    // storage. To fix this, we recompute next_write_ts here, and close the file if it doesn't.
```

Comment:
That's true for storage type "NumSnaps". But with type "one_month" (say), we can't say if the file is full unless we know the time stamp of the next write.

Reply:
We cannot compute the next time stamp during t0 output. The driver does not have info about `dt` during the init sequence, which is when the OM is setup. When I originally designed the driver, I wanted to separate concerns as much as possible. In my mind, dt was a "run" time param, not an "init" time param (I didn't even know if dt could in principle change dynamically down the road).

If you want to compute `next_write_ts` during t0 output (which, again, happens during the init sequence), we need to pass dt to the driver init methods (from the f90 cpl interface). We can of course do that. And all in all, it may make the code simpler. It's a slightly deeper interface change though, so we could do it as a follow up PR.